optuna/visualization.py

```diff
@@ -98,9 +98,9 @@ def _get_intermediate_plot(study):
     if len(trials) == 0:
         logger.warning('Study instance does not contain trials.')
         return go.Figure(data=[], layout=layout)
-    if hasattr(trials[0], 'intermediate_values') is False:
+    if any([len(t.intermediate_values) == 0 for t in trials]):
```

Comment:
N.B. It might be better to use `all` here instead.

Reply:
I agree with you. Feel free to fix this code within this PR or not.